add 3 font rendering modes `FontRenderMode`, preserve global option a…

bung87 opened the pull request:
…ffect `frmDefault`

Behavior with `opts.setEmbedFont(false)` in demo:**

| Local `renderMode` | Final Result | Behavior |
|-------------------|--------------|----------|
| `frmDefault` | `frmDefault` | No embedding, standard text rendering |
| `frmEmbed` | `frmEmbed` | **Individual choice overrides global** - font embedded |
| `frmPathRendering` | `frmPathRendering` | **Not affected by global setting** - text as paths |

### **Behavior if you changed to `opts.setEmbedFont(true)`:**

| Local `renderMode` | Final Result | Behavior |
|-------------------|--------------|----------|
| `frmDefault` | `frmEmbed` | **Global forces embedding for defaults** |
| `frmEmbed` | `frmEmbed` | Same result |
| `frmPathRendering` | `frmPathRendering` | Still renders as paths, not affected |

jangko committed:
Merge pull request #16 from windgo-inc/will-devel

Updated to support more images, added fallback to stb_image

jangko committed:
Merge branch 'master' into new_feature

jangko committed:
add migration.md

jangko committed:
fix some API

jangko committed:
fix demo

jangko committed:
fix some bugs

jangko committed:
update migration.md

jangko committed:
more api fix

jangko committed:
fix wtf8

jangko committed:
update .nimble

jangko committed:
compiles with latest nim

jangko committed:
fixes import

jangko committed:
fixes #17

jangko committed:
fixes .nimble

jangko committed:
fix build script

jangko committed:
Merge pull request #18 from jangko/fixbuild

fix build script

jangko committed:
update readme.md

jangko committed:
fix module importing problem

jangko committed:
Merge pull request #21 from jangko/fix20

fix module importing problem

jangko committed:
fixes page.nim compilation error

jangko committed:
attempt to make test green

jangko committed:
remove compiler warnings

jangko committed:
Merge pull request #22 from jangko/fix_ci

make test green

jangko committed:
override nim 1.2 side effect analysis

jangko committed:
Merge pull request #25 from jangko/nim1.2

fixes #24, override nim 1.2 side effect analysis

jangko committed:
upgrade mingw gcc v4.9.2 to v8.1.0 in appveyor

jangko committed:
fix #26, deals with holed enum in a for loop

jangko committed:
fix #27, wrong base14 text height and TTF text height crash

jangko committed:
fix .nimble

jangko committed:
Merge pull request #28 from jangko/various_fix

fix #26, fix #27

jangko committed:
fix widget creation in nimPDF.nim

jangko committed:
update demo/pushButton.pdf

jangko committed:
fix text_field

jangko committed:
fix push button

jangko committed:
fix appveyor.yml

jangko committed:
Merge pull request #31 from jangko/fix_widget

Fix widget

jangko committed:
add annotUri

jangko committed:
fix annot_uri demo

jangko committed:
Merge pull request #32 from jangko/annot_uri

add annotUri

jangko committed:
add github action

jangko committed:
Merge pull request #33 from jangko/github_action

add github action

jangko committed:
add msvc/vcc compiler test

jangko committed:
fixes github action script

ThomasTJdev committed:
Make PDF objects public to allow access to pages

ThomasTJdev committed:
fix typo

ThomasTJdev committed:
Use public procs for navigating between pages

ThomasTJdev committed:
pass around 'embedFonts` and pre-cache charmap

jangko committed:
Fix CI (#46)

jangko committed:
Remove travis.yml and appveyor.yml (#47)

* Remove travis.yml and appveyor.yml

* Remove travis and appveyor badge from readme.md

ThomasTJdev committed:
fix wrong width with some fonts

bung87 committed:
font embedding

bung87 committed:
Remove demo/test_embedding_size.nim from version control

bung87 committed:
`drawTextAsPaths` if not embed

ThomasTJdev committed:
Merge pull request #1 from bung87/embed-fonts2

Embed fonts

ThomasTJdev committed:
Merge branch 'master' into feature/ttj/option-to-embed-fonts

bung87 committed:
add 3 font rendering modes `FontRenderMode`, preserve global option affect `frmDefault`

ThomasTJdev commented:
Looks good @bung87 - I like it with enums. However, I can select the `frmPathRendering`:
<img width="928" height="783" alt="image" src="https://github.com/user-attachments/assets/117d445e-7514-4dc0-b449-6b8222b0f544" />

bung87 committed:
Remove embedFont option from PDFOptions and related functionality

- Removed embedFont field from PDFOptions type in options.nim
- Removed setEmbedFont, getEmbedFont procedures from options.nim
- Updated page.nim to remove global embedFont dependency in font rendering
- Removed Example 7 from embed_fonts.nim demo that showed global override behavior
- Updated documentation comments to reflect removal of global embedding option
- Font embedding behavior now controlled entirely by renderMode parameter

bung87 committed:
Clean up repository by removing binary files and generated artifacts

- Removed .DS_Store files (macOS system files)
- Removed compiled executables from demo directory
- Removed generated PDF files from demo directory
- These files should not be tracked in git history

bung87 committed:
Update .gitignore to prevent tracking of binary files and generated artifacts

- Added patterns for .DS_Store files (macOS system files)
- Added patterns for compiled executables and demo PDFs
- Added patterns for build artifacts and cache directories
- Prevents future accidental commits of generated files

bung87 committed:
Merge branch 'feature/ttj/option-to-embed-fonts' into embed-fonts2

bung87 committed:
update .gitignore

bung87 committed:
clean up .gitignore

ThomasTJdev commented:
Hi Bung,

Please see below when I'm running the embed_font demo file - it uses the Eunjin font.

**Default file**
<img width="1895" height="619" alt="image" src="https://github.com/user-attachments/assets/49104c65-b0f5-4b11-824d-b71a39d74b1b" />


**Both set to default**
<img width="1895" height="619" alt="image" src="https://github.com/user-attachments/assets/8460329b-f4ca-46ee-aa8c-2ddb96bad591" />


**Both to pathRendering - then it's not embedded**
<img width="1895" height="619" alt="image" src="https://github.com/user-attachments/assets/31fb06fc-133f-4456-8acf-13b61371bea5" />

bung87 committed:
Fix frmDefault font rendering to properly disable embedding

- Fixed TTFont EscapeString method to respect renderMode parameter correctly
- Updated demo comments and output to clarify frmDefault means no embedding
- frmDefault now properly renders text without font embedding as intended
- Resolves issue where frmDefault was incorrectly embedding fonts

bung87 committed:
remove global option
